tests/integration/transactions/test_amm_bid.py

```diff
@@ -0,0 +1,142 @@
+from tests.integration.integration_test_case import IntegrationTestCase
+from tests.integration.it_utils import (
+    sign_and_reliable_submission_async,
+    test_async_and_sync,
+)
+from tests.integration.reusable_values import AMM, WALLET
+from xrpl.models.amounts.issued_currency_amount import IssuedCurrencyAmount
+from xrpl.models.auth_account import AuthAccount
+from xrpl.models.requests.amm_info import AMMInfo
+from xrpl.models.transactions.amm_bid import AMMBid
+from xrpl.models.transactions.amm_deposit import AMMDeposit, AMMDepositFlag
+
+
+class TestAMMBid(IntegrationTestCase):
+    @test_async_and_sync(globals())
+    async def test_basic_functionality(self, client):
+        asset = AMM["asset"]
+        asset2 = AMM["asset2"]
+
+        response = await sign_and_reliable_submission_async(
+            AMMDeposit(
+                account=WALLET.classic_address,
+                asset=asset,
+                asset2=asset2,
+                amount="1000",
+                flags=AMMDepositFlag.TF_SINGLE_ASSET,
+            ),
+            WALLET,
+            client,
+        )
+
+        self.assertTrue(response.is_successful())
+        self.assertEqual(response.result["engine_result"], "tesSUCCESS")
+
+        pre_amm_info = await client.request(
+            AMMInfo(
+                asset=asset,
+                asset2=asset2,
+            )
+        )
+
+        response = await sign_and_reliable_submission_async(
+            AMMBid(
+                account=WALLET.classic_address,
+                asset=asset,
+                asset2=asset2,
+            ),
+            WALLET,
+            client,
+        )
+
+        self.assertTrue(response.is_successful())
+        self.assertEqual(response.result["engine_result"], "tesSUCCESS")
+
+        amm_info = await client.request(
+            AMMInfo(
+                asset=asset,
+                asset2=asset2,
+            )
+        )
+
+        self.assertGreater(
+            float(amm_info.result["amm"]["auction_slot"]["price"]["value"]),
+            float(pre_amm_info.result["amm"]["auction_slot"]["price"]["value"]),
+        )
+        self.assertLess(
+            float(amm_info.result["amm"]["lp_token"]["value"]),
+            float(pre_amm_info.result["amm"]["lp_token"]["value"]),
+        )
```

Comment:
Why is this not asserting exact values?

Reply:
2 reasons:
- The math to compute/predict the result values is complex and their result computations are different in JavaScript vs C++.
- One AMM pool is being shared among all tests so the values are constantly changing

Thus given these conditions, rather than asserting exact values, I think it's sufficient to just check whether tokens were added/removed after each transaction. Would be ideal if there was a rippled API to give the result exact values but there isn't.

Reply:
Shouldn't the difference results still be determined in a test, since the numbers can all be hardcoded?

Reply:
Looks like they can't be hardcoded to a single value since each test runs twice (sync/async). This results in differing values:
```
AssertionError: 3311.0 != 1311
```
```
AssertionError: 4311.0 != 1311
```

Reply:
Is the difference not constant? Seems like it is in this example.

Reply:
It's not constant, the left hand value changes from `3311` to `4311`

Reply:
Yes, I mean the amount added/subtracted (from the value before the transaction was sent)

Reply:
Ah I see what you mean :) Yes, looks like it